assets/js/modules/analytics-4/components/common/PropertySelect.test.js

```diff
@@ -0,0 +1,124 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import PropertySelect from './PropertySelect';
+import { MODULES_ANALYTICS, ACCOUNT_CREATE } from '../../../analytics/datastore/constants';
+import { STORE_NAME } from '../../datastore/constants';
+import { CORE_SITE } from '../../../../googlesitekit/datastore/site/constants';
+import * as fixtures from '../../datastore/__fixtures__';
+import * as analyticsFixtures from '../../../analytics/datastore/__fixtures__';
+import { fireEvent, act, render, muteFetch } from '../../../../../../tests/js/test-utils';
+
+const {
+	createProperty,
+	createWebDataStream,
+	properties,
+	webDataStreams,
+} = fixtures;
+const { accounts } = analyticsFixtures.accountsPropertiesProfiles;
+const accountID = createProperty._accountID;
+const propertyID = createWebDataStream._propertyID;
+
+const setupRegistry = ( { dispatch } ) => {
+	dispatch( CORE_SITE ).receiveSiteInfo( { referenceSiteURL: 'http://example.com' } );
+	dispatch( MODULES_ANALYTICS ).receiveGetSettings( {} );
+	dispatch( MODULES_ANALYTICS ).setAccountID( accountID );
+
+	dispatch( MODULES_ANALYTICS ).receiveGetAccounts( accounts );
+	dispatch( MODULES_ANALYTICS ).finishResolution( 'getAccounts', [] );
+
+	dispatch( STORE_NAME ).receiveGetProperties( properties, { accountID } );
+	dispatch( STORE_NAME ).finishResolution( 'getProperties', [ accountID ] );
+
+	dispatch( STORE_NAME ).receiveGetWebDataStreams( webDataStreams, { propertyID } );
+	dispatch( STORE_NAME ).finishResolution( 'receiveGetWebDataStreams', { propertyID } );
+};
+
+const setupEmptyRegistry = ( { dispatch } ) => {
+	dispatch( MODULES_ANALYTICS ).setSettings( {} );
+	dispatch( MODULES_ANALYTICS ).setAccountID( accountID );
+
+	dispatch( MODULES_ANALYTICS ).receiveGetAccounts( accounts );
+	dispatch( MODULES_ANALYTICS ).finishResolution( 'getAccounts', [] );
+
+	dispatch( STORE_NAME ).receiveGetProperties( [], { accountID } );
+	dispatch( STORE_NAME ).finishResolution( 'getProperties', [ accountID ] );
+};
+
+const setupRegistryWithNoAccountID = ( registry ) => {
+	setupRegistry( registry );
+	registry.dispatch( STORE_NAME ).finishResolution( 'getProperties', [ ACCOUNT_CREATE ] );
+};
+
+describe( 'PropertySelect', () => {
+	it( 'should render an option for each analytics property of the currently selected account.', async () => {
+		muteFetch( /^\/google-site-kit\/v1\/modules\/analytics-4\/data\/settings/, [] );
```

Comment:
If I remove the `muteFetch` the tests start failing with e.g.
```console
["Google Site Kit API Error", "method:GET", "datapoint:settings", 
"type:modules", "identifier:analytics-4", "error:\"window.fetch is not a function\""]
```
So things are indeed calling `fetch` under the hood somewhere to get at the `settings` endpoint. I'm not sure where this is happening. Is there something I'm doing wrong in `setupRegistry` perhaps?

Reply:
Ah that helps. It's because you're only receiving settings for `modules/analytics` not GA4 but it uses both so it's trying to fetch the settings for the GA4 module.

Reply:
Bingo. Thanks - I've removed `muteFetch` now 👍